### PR TITLE
halide: remove mesa dependency on darwin

### PR DESCRIPTION
**Pull request by twesterhout**
###### Description of changes

- Remove `mesa` dependency for unless we're on linux.
- For linux, we also need `libGL` such that CMake can find OpenGL.
- Add `libffi` because CMake's configure step complained about it not being found.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/halide/default.nix [hunk 1]
```diff
@@ -43,11 +45,14 @@ stdenv.mkDerivation rec {
     llvmPackages.lld
     llvmPackages.openmp
     llvmPackages.libclang
+    libffi
     libpng
     libjpeg
-    mesa
     eigen
     openblas
+  ] ++ lib.optionals stdenv.isLinux [
```
Comment (wegank) [inline pkgs/development/compilers/halide/default.nix:53]:
```suggestion
  ] ++ lib.optionals (!stdenv.isDarwin) [
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -16598,6 +16598,12 @@
     githubId = 9413924;
     name = "Thorsten Weber";
   };
+  twesterhout = {
```
Comment (wegank) [inline maintainers/maintainer-list.nix:16601]: Would you please do this in a separate commit titled `maintainers: add twesterhout`? Thanks!

Comment (twesterhout) [reply maintainers/maintainer-list.nix:16601]: Good point! Done.

**Review by wegank — APPROVED** [review]
